clarity/src/vm/analysis/type_checker/mod.rs

```diff
@@ -192,6 +192,29 @@ impl FunctionType {
                 }
                 Err(CheckErrors::UnionTypeError(arg_types.clone(), found_type.clone()).into())
             }
+            FunctionType::ArithmeticBinaryUIntAsSecondArg => {
+                check_argument_count(2, args)?;
+                let (first, second) = (&args[0], &args[1]);
+                analysis_typecheck_cost(accounting, &TypeSignature::IntType, first)?;
+                analysis_typecheck_cost(accounting, &TypeSignature::IntType, second)?;
```

Comment:
This should be `analysis_typecheck_cost(accounting, &TypeSignature::UIntType, second)?;`, right?

Reply:
Most likely :) Updated this now.  Would you be able to describe a little what the intended usage of `analysis_typecheck_cost` is? I tried to infer its usage by looking at the other function types - it looks like it's used to evaluate the type checking cost of each input parameter? Would that be a correct guess?

Reply:
It assesses the runtime cost of performing the type-checking on the second argument.  Analyzing a smart contract isn't free; it consumes a block's compute budget ("gas").